src/python/pants/java/nailgun_protocol.py

```diff
@@ -253,3 +258,12 @@ def str_int_bool(i):
       return i.isdigit() and bool(int(i))  # Environment variable values should always be strings.
 
     return tuple(str_int_bool(env.get(cls.TTY_ENV_TMPL.format(fd_id), '0')) for fd_id in range(3))
+
+  @classmethod
+  def ttynames_from_env(cls, env):
+    """Determines the ttynames for remote file descriptors (if ttys).
+
+    :param dict env: A dictionary representing the environment.
+    :returns: A tuple of boolean values indicating ttyname paths or None for (stdin, stdout, stderr).
+    """
+    return tuple(env.get(cls.TTY_PATH_ENV.format(fd_id)) for fd_id in range(3))
```

Comment:
I'd be slightly happier with `for fd_id, fd in ((0, stdin), (1, stdout), (2, stderr)):` above and `... for fd_id in (0, 1, 2))` here just because 0, 1, & 2 are special and the lucky behavior of enumerate and range (0-based sequence generators), masks this a bit. That said, `range(3)` has precedent in this file.

Reply:
works for me - fixed.